### PR TITLE
Add model ids

### PR DESCRIPTION
**Pull request by juxd**
tests are inconsistent for some reason.

please help me do your tests too!

### CODE REVIEW

File: src/test/data/XmlSerializableRouteListTest/invalidRouteRouteList.xml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
 <routelist>
     <!-- Route with invalid destination field -->
-    <routes>
+    <routes id="0">
```
Comment (jinyingtan) [inline src/test/data/XmlSerializableRouteListTest/invalidRouteRouteList.xml]: Is 0 an invalid `id` for routes?

File: src/main/java/seedu/address/model/deliveryman/Deliveryman.java [hunk 2]
```diff
@@ -4,17 +4,32 @@
...
-public class Deliveryman {
+public class Deliveryman extends IdObject {
+    /** Tracker for assigning id. */
+    private static int idCounter = 0;
+
     private final Name name;
     //TODO:  add field for orders/routes attached to deliveryman.
 
     public Deliveryman(Name name) {
+        super(idCounter++);
```
Comment (kohchihao) [inline src/main/java/seedu/address/model/deliveryman/Deliveryman.java]: Consider using `ModelManager` to increment the `idCounter`. So you can just pass in only `id` and `name`.

File: src/main/java/seedu/address/model/order/Order.java [hunk 3]
```diff
@@ -27,6 +30,22 @@
      * Every field must be present and not null.
      */
     public Order(Name name, Phone phone, Address address, OrderDate orderDate, Set<Food> food) {
+        super(idCounter++);
```
Comment (kohchihao) [inline src/main/java/seedu/address/model/order/Order.java]: similar to the previous comment.

File: src/main/java/seedu/address/model/route/Route.java [hunk 4]
```diff
@@ -23,12 +27,31 @@
      * @param destination The address of the restaurant.
      */
     public Route(Address destination) {
+        super(idCounter++);
```
Comment (kohchihao) [inline src/main/java/seedu/address/model/route/Route.java]: similar to the previous comment

File: src/main/java/seedu/address/model/IdObject.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
+package seedu.address.model;
+
+import java.util.UUID;
+
+/**
+ * Represents the Objects that need an ID for storage & reference
+ */
+public abstract class IdObject {
+    public static final String MESSAGE_INVALID_ID = "Not an valid ID! Please check SML File.";
```
Comment (kohchihao) [inline src/main/java/seedu/address/model/IdObject.java]:
```suggestion
    public static final String MESSAGE_INVALID_ID = "Not an valid ID! Please check XML File.";
```

**Pull request merged**